tsdb/engine/tsm1/mmap_unix.go

```diff
@@ -19,3 +20,12 @@ func mmap(f *os.File, offset int64, length int) ([]byte, error) {
 func munmap(b []byte) (err error) {
 	return syscall.Munmap(b)
 }
+
+func madvise(b []byte, advice int) error {
+	// For some reason madvise is not in the stdlib for darwin
+	_, _, errno := syscall.Syscall(syscall.SYS_MADVISE, uintptr(unsafe.Pointer(&b[0])), uintptr(len(b)), uintptr(advice))
```

Comment:
Further development in the `syscall` package was locked down by the Go team and they recommend all 3rd parties use `golang.org/x/sys` package. It is noted on the [syscall package overview help](https://golang.org/pkg/syscall/):

> NOTE: This package is locked down. Code outside the standard Go repository should be migrated to use the corresponding package in the golang.org/x/sys repository.

[Madvise](https://godoc.org/golang.org/x/sys/unix#Madvise) does exist in `golang.org/x/sys/unix` and there is an [implementation for Darwin](https://github.com/golang/sys/blob/master/unix/zsyscall_darwin_amd64.go#L269). Should we switch to this this package?

Reply:
Done.